symbulate/tests/test_distributions.py

```diff
@@ -17,18 +17,460 @@ def test_p_one(self):
+        X = 10 + 5 * Z
+        sims = X.sim(Nsim)
+        cdf = stats.norm(loc = 10, scale = 5).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+    def test_normal_to_Gamma(self):
+        X = RV(Normal(0, 1))
+        X = X ** 2
+        sims = X.sim(Nsim)
+        cdf = stats.gamma(a = 1/2, scale = 2).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+    def test_normal_to_ChiSquare(self):
+        X = RV(Normal(0, 1))
+        X = X ** 2
+        sims = X.sim(Nsim)
+        cdf = stats.chi2(df = 1).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+
+class TestExponential(unittest.TestCase):
+    
+    def test_Exponential_error(self):
+        self.assertRaises(Exception, lambda: Exponential(-5))
+
+    def test_Exponential_to_Gamma(self):
+        X = RV(Exponential(rate = 0.9))
+        sims = X.sim(Nsim)
+        exp = stats.gamma(scale = 1/0.9, a = 1).cdf
+        pval = stats.kstest(sims, exp).pvalue
+        self.assertTrue(pval > 0.01)
+
+    def test_Exponential_sum_Gamma(self):
+        X,Y,Z,A = RV(Exponential(rate = 0.9) ** 4)
+        sims = (X+Y+Z+A).sim(Nsim)
+        exp = stats.gamma(scale = 1/0.9, a = 4).cdf
+        pval = stats.kstest(sims, exp).pvalue
+        self.assertTrue(pval > 0.01)
+
+
+class TestGamma(unittest.TestCase):
+
+    def test_Gamma_shape_error(self):
+        self.assertRaises(Exception, lambda: Gamma(shape = -5, rate = 40))
+
+    def test_Gamma_rate_error(self):
+        self.assertRaises(Exception, lambda: Gamma(shape = 4, rate = -10))
+
+    def test_Gamma_to_Exponential(self):
+        X = Gamma(shape = 1, rate = 1/ 0.9)
+        sims = X.sim(Nsim)
+        exp = stats.expon(scale = 0.9).cdf
+        pval = stats.kstest(sims, exp).pvalue
+        self.assertTrue(pval > .01)
+
+        
+    def test_Gamma_reshape(self):
+        def g(x):
+            return x*8
+
+        X = RV(Gamma(shape = 9, scale = 4))
+        sims = X.sim(Nsim).apply(g)
+        exp = stats.gamma(scale = 4 * 8, a = 9).cdf
+        pval = stats.kstest(sims, exp).pvalue
+        self.assertTrue(pval > .01)
+ 
+    def test_Gamma_additive(self):
+        X,Y = RV(Gamma(shape = 10, scale = 0.5) * Gamma(shape = 8, scale = 0.5))
+        sims = (X+Y).sim(Nsim)
+        exp = stats.gamma(scale = 0.5, a = 18).cdf
+        pval = stats.kstest(sims, exp).pvalue
+        self.assertTrue(pval > .01)
+
+
+class TestBeta(unittest.TestCase):
+    
+    def test_Beta_error_a(self):
+        self.assertRaises(Exception, lambda: Beta(a = -10, b = 3))
+
+    def test_Beta_error_b(self):
+        self.assertRaises(Exception, lambda: beta(a = 3, b = -10))
+
+    def test_Beta_to_Uniform(self):
+        X = Beta(a = 1, b = 1)
+        sims = X.sim(Nsim)
+        exp = stats.uniform(0, 1).cdf
+        pval = stats.kstest(sims, exp).pvalue
+        self.assertTrue(pval > .01)
+
+    def test_Beta_to_ChiSquare(self):
```

Comment:
I don't understand this test; it seems to have nothing to do with the Beta distribution.